src/main/java/teammates/common/datatransfer/GeneralLogEntry.java

```diff
@@ -0,0 +1,114 @@
+package teammates.common.datatransfer;
+
+import java.util.Map;
+import java.util.Objects;
+
+import javax.annotation.Nullable;
+
+/**
+ * This class represents a log entry and contains some of the fields that are more important
+ * for querying logs action and are of more interest to maintainers.
+ */
+public class GeneralLogEntry {
+    private final String logName;
+    private final String severity;
+    private final String trace;
+    private final SourceLocation sourceLocation;
+    private final long timestamp;
+    @Nullable
+    private String textPayloadMessage;
```

Comment:
`textPayload` is to some extent Cloud Logging's terminology. What this really means is just log message.